### PR TITLE
Implemented ToastNotification + unit tests

### PR DESCRIPTION
**Pull request by danielayala94**
This is the implementation of ToastNotification class.

Also added several unit tests to verify the toast properties in packaged and unpackaged scenarios.

### CODE REVIEW

File: test/TestApps/ToastNotificationsTestApp/ToastNotificationsTestApp.cpp [hunk 1]
```diff
@@ -186,6 +188,83 @@ bool VerifyFailedToastAssetsWithNullIconPath_Unpackaged()
...
+    winrt::hstring group{ L"group" };
+    toast.Group(group);
+
+    if (toast.Group() != group)
+    {
+        return false;
+    }
+
+    /*
+    * TODO: Uncomment once ToastProgressData has been implemented
```
Comment (sharath2727) [inline test/TestApps/ToastNotificationsTestApp/ToastNotificationsTestApp.cpp:222]: Thanks.

File: test/TestApps/ToastNotificationsTestApp/ToastNotificationsTestApp.cpp [hunk 2]
```diff
@@ -186,6 +188,83 @@ bool VerifyFailedToastAssetsWithNullIconPath_Unpackaged()
     return false;
 }
 
+bool VerifyToastProperties()
```
[the remark this answers is not in the document]
Comment (sharath2727) [reply test/TestApps/ToastNotificationsTestApp/ToastNotificationsTestApp.cpp:191]: Yeah I just think this is a big test. We can break it down I guess.

**Review by sharath2727 — CHANGES_REQUESTED** [review]

File: test/TestApps/ToastNotificationsTestApp/ToastNotificationsTestApp.cpp [hunk 3]
```diff
@@ -186,6 +188,83 @@ bool VerifyFailedToastAssetsWithNullIconPath_Unpackaged()
...
+    expirationTime += winrt::TimeSpan(std::chrono::seconds(10));
+
+    toast.ExpirationTime(expirationTime);
+    if (toast.ExpirationTime() != expirationTime)
+    {
+        return false;
+    }
+
+    toast.Priority(winrt::ToastPriority::High);
+    if (toast.Priority() != winrt::ToastPriority::High)
```
Comment (loneursid) [inline test/TestApps/ToastNotificationsTestApp/ToastNotificationsTestApp.cpp:248]: Should we verify that the value is set the default before verifying that the value matches the value we just set? 

Same questions to most of the tests in test method.

File: test/TestApps/ToastNotificationsTestApp/ToastNotificationsTestApp.cpp [hunk 4]
```diff
@@ -186,6 +188,83 @@ bool VerifyFailedToastAssetsWithNullIconPath_Unpackaged()
...
+    toast.ProgressData(progressData);
+
+    auto progressDataFromToast = toast.ProgressData();
+    if (progressDataFromToast != progressData)
+    {
+        return false;
+    }
+    */
+
+    winrt::DateTime expirationTime = winrt::clock::now();
```
[the remark this answers is not in the document]
Comment (danielayala94) [reply test/TestApps/ToastNotificationsTestApp/ToastNotificationsTestApp.cpp]: Done there and in all applicable places under PR scope.

**Review by hulumane — APPROVED** [review]

**Pull request merged**